src/autoscaler/metricscollector/collector/collector.go

```diff
@@ -196,12 +195,15 @@ func (c *Collector) SaveMetricsInDB() {
 		case metric := <-c.dataChan:
 			metrics = append(metrics, metric)
 		case <-ticker.C():
-			go func(instancemetricsDb db.InstanceMetricsDB, metrics []*models.AppInstanceMetric) {
-				instancemetricsDb.SaveMetricsInBulk(metrics)
+			if c.persistMetrics && len(metrics) > 0 {
```

Comment:
will empty metrics slice cause issue?

Reply:
@boyang9527 
Do you mean the following condition:

```
package main
import (
	"fmt"
)
func main() {
	var emptySlice []int = []int{}
	var nilSlice []int = nil
	fmt.Printf("the length of empty slice is %d\n", len(emptySlice))
	fmt.Printf("the leng of nil slice is %d\n", len(nilSlice))

}
```
The result is :
```
the length of empty slice is 0
the leng of nil slice is 0
```

Reply:
I guess @boyang9527  is just asking why the `len()>0` check is added here.  
In fact, just to avoid the necessary db connection/transition when there is nothing to save.

Reply:
@cdlliuy yes. I have checked with him.